File: src/algoexpert/closestBst.js
function findClosestValueInBst(tree, target) {
  return closestHelper(tree, target, tree.value);
}

function closestHelper(tree, target, closest) {
  if (tree === null) return closest;

  if (Math.abs(target - closest) > Math.abs(target - tree.value))
    closest = tree.value;

  if (target < tree.value) return closestHelper(tree.left, target, closest);
  if (target > tree.value) return closestHelper(tree.right, target, closest);
  return closest;
}

class BST {
  constructor(value) {
    this.value = value;
    this.left = null;
    this.right = null;
  }
}

File: src/algoexpert/closestBst.py
# O(log(n))t | O(log(n))s
def findClosestValueInBst(tree, target):
   return bstHelper(tree, target, tree.value)

def bstHelper(tree, target, closest):
	if tree is None:
		return closest

	if abs(target - closest) > abs(tree.value - target):
		closest = tree.value

	if target < tree.value:
		return bstHelper(tree.left, target, closest)
	elif target > tree.value:
		return bstHelper(tree.right, target, closest)
	else:
		return closest
